### DISCUSSION TITLE
Orientation marker

### DISCUSSION THREAD

**Discussion by MarkHaday** [ImageMagick/ImageMagick#4773] [Digital Image Processing]
We scan in large amounts of documents that contain QR codes in different places.  We use Zbar to read to QR code.  Is there anyway we can detect the orientation via IM in case the document has been scanned in upside down?  To be clear, this is not a simple case of landscape vs portrait, but upside down portrait.

**Comment by fmw42** [comment]
try reading with tesseract. If that fails, then likely upside down. So try rotating 180 and reading again with tesseract.  ImageMagick has no text reading capabilities. It can just convert ascii text to raster images.

**Reply by MarkHaday** [reply]
Thank you.  I am playing the with tesseract orientation and that seems to work.  It has and 'Orientation confidence' and will just have to try it out
